Z/Zlibcloudflare/build_tarballs.jl

```diff
@@ -0,0 +1,40 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Zlibcloudflare"
+version = v"1.2.8"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/cloudflare/zlib.git", "959b4ea305821e753385e873ec4edfaa9a5d49b7")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/zlib*
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DUNIX=true -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DBUILD_SHARED_LIBS=ON
+make -j ${nproc}
+make install
+install_license ../README
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "musl")
+]
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libz", :libz)
```

Comment:
I can't say I like having multiple libraries around with same name (and likely soname?), point which I made repeatedly in #3801.  That's going to be brittle, to say the least.

Reply:
I'm fine with renaming it since I think few ppl would use it

Reply:
Do you plan to link other libraries to this or you just want to `ccall` into this library directly?

Reply:
I probably need to copy the source code of https://github.com/JuliaIO/CodecZlib.jl

Reply:
When trying to build for aarch64:
```
-- Performing Test ARM_CRC
-- Performing Test ARM_CRC - Failed
-- Performing Test HAS_SSE2
-- Performing Test HAS_SSE2 - Success
-- Performing Test HAS_SSSE3
-- Performing Test HAS_SSSE3 - Success
-- Performing Test HAS_SSE42
-- Performing Test HAS_SSE42 - Success
-- Performing Test HAS_PCLMUL
-- Performing Test HAS_PCLMUL - Success
```
sigh... :facepalm:

Reply:
Judging by the plethora of `-march` and other `-m` options they really like to use, this might be a good exercise for our totally-untested-in-the-wild multiarchitecture infrastructure :upside_down_face:

Reply:
Renaming the variable `:libz` -> `:libzcloudfare` is hardly a problem at all  :slightly_smiling_face:   My suggestion was to rename the library, with, most importantly, its SONAME everywhere in the build system and inside the pkgconfig file, which should also be renamed :slightly_smiling_face:  Unless you can show that you can build the JLL and call into the right libz as it is now (which I have _**many**_ doubts with).

Reply:
since the output is a single SO file, can we just rename it after `make`?

Reply:
As I'm trying to say repeatedly here and in #3801, I'm worried about the [soname](https://en.wikipedia.org/wiki/Soname), not just the filename itself.  And do that consistently you must fix the soname _**everywhere**_ in the build system.

Reply:
sorry, not familiar with ABI/linker, guess this is too much patching work for too little gain. Thanks for the explanation.